### PR TITLE
Gc intro block

### PR DESCRIPTION
**Pull request by jmealing**
To do:

[ ✓] GCWeb french
[✓] DTO guidance

DTO design system PRs:
EN: https://github.com/canada-ca/design-system/pull/307
FR: https://github.com/canada-ca/systeme-conception/pull/210

### CODE REVIEW

File: components/gc-intro/gc-intro-block-en.html
Comment (delisma) [inline components/gc-intro/gc-intro-block-en.html:10]:
```suggestion
	<h1 property="name" id="wb-cont" dir="ltr">{{ page.title }}</h1>
```

**Review by delisma — CHANGES_REQUESTED** [review]

File: components/gc-intro/gc-intro-block-en.html
Comment (delisma) [inline components/gc-intro/gc-intro-block-en.html:13]:
```suggestion
	<p>The introduction block pattern introduces the content of a landing page.</p>
```

File: components/gc-intro/gc-intro-block-en.html
Comment (delisma) [inline components/gc-intro/gc-intro-block-en.html:24]:
```suggestion

```

File: components/gc-intro/gc-intro-block-en.html
Comment (delisma) [inline components/gc-intro/gc-intro-block-en.html:27]:
```suggestion

```

File: components/gc-intro/gc-intro-block-en.html
Comment (delisma) [inline components/gc-intro/gc-intro-block-en.html:36]:
```suggestion

```

File: components/gc-intro/gc-intro-block-en.html
Comment (delisma) [inline components/gc-intro/gc-intro-block-en.html:46]:
```suggestion
	</ul>
```

**Review by delisma — CHANGES_REQUESTED** [review]

**Comment by duboisp** [comment]
Pre-approved upon successful local testing and review.

File: components/gc-intro/gc-intro-block-en.html
Comment (duboisp) [inline components/gc-intro/gc-intro-block-en.html:60]: This do create an heading structure issue.

There is a few alternative like informing the user of that irregular page structure or simpler will be to demo it from it's own dedicate page.

I do recommend to have the demo on a separate page and keep the source code here.

File: components/gc-intro/gc-intro-block-en.html
Comment (duboisp) [inline components/gc-intro/gc-intro-block-en.html:38]:
```suggestion
	<p>The introduction block pattern consists of, as demoed on this documentiton page:</p>
```

File: components/gc-intro/gc-intro-block-en.html
Comment (duboisp) [inline components/gc-intro/gc-intro-block-en.html:43]:
```suggestion
	</ul>
	<pre><code>
	&lt;h1 property="name" id="wb-cont" dir="ltr">Introduction block pattern&lt;/h1>
	&lt;p>The introduction block pattern introduces the content of a landing page.&lt;/p>
	</code></pre>
	<p>It can also include:</p>
```

File: components/gc-intro/gc-intro-block-en.html
Comment (duboisp) [inline components/gc-intro/gc-intro-block-en.html:78]: Should we also add the introduction block pattern that contain the by-line?

ref. https://design.canada.ca/common-design-patterns/institutional-byline.html

**Review by duboisp — CHANGES_REQUESTED** [review]
A few changes:
* [ ] The irregular heading structure need to be addressed either by proper messaging or by moving the demo onto its own page.
* [ ] Missing a demo for the first example, I proposed some text and to reuse the current page as being the demo page
* [ ] Should the intro block with the byline added?

Side question: We should think of how this intro pattern are going to be linked with the heading level 1 component (which include double h1). 
Is this design pattern are only informational and suggestive? Or will it needs to be monitored like other stable component, to advise web author of any update that must be applied to the intro block if there is a major change?

File: components/gc-intro/gc-intro-block-en.html
Comment (jmealing) [reply components/gc-intro/gc-intro-block-en.html:78]: The byline is being moved to the bottom of the page, like how it is in the theme and topic beta, so it doesn't make sense as part of this component.

File: components/gc-intro/gc-intro-block-en.html
Comment (jmealing) [reply components/gc-intro/gc-intro-block-en.html:60]: Do you have an example I can use for creating separate demo page? Wouldn't it have the same issue though of two h1s on the page because the demo page will still have an h1

File: components/gc-intro/gc-intro-block-en.html
Comment (jmealing) [reply components/gc-intro/gc-intro-block-en.html:43]: If we do the demo on a separate page, it seems we don't need this edit?

File: components/gc-intro/gc-intro-block-en.html
Comment (jmealing) [reply components/gc-intro/gc-intro-block-en.html:38]: I will add a demo link below the bullets going to the demo page with all the examples once I create it instead of adding text here.

File: components/gc-intro/gc-intro-block-en.html
Comment (duboisp) [reply components/gc-intro/gc-intro-block-en.html:60]: For that you will need to create a new .html file in the folder. I suggest to use the layout "no-container" for this example.

That page markup could look like this:
```
---
{
	"title": "GC intro block with super task",
	"language": "fr",
	"altLangPage": "intro-block-fr.html",
	"layout": "no-container",
	"dateModified": "2023-09-13"
}
---
<div class="bg-center bg-cover" data-bgimg-srcset="img/1x1.png 991w, img/ip-cover-image-1200x726.jpg 992w">
	<div class="container">
		<div class="row mrgn-bttm-md">
			<div class="col-md-7">
			<h1 property="name" id="wb-cont">[Institution name]</h1>
				<p>Short description of the institution’s mandate.</p>
				<p><a class="btn btn-call-to-action" href="#">Super task button [optional]</a></p>
			</div>
		</div>
	</div>
</div>

<pre><code>&lt;div class="bg-center bg-cover" data-bgimg-srcset="img/1x1.png 991w, img/ip-cover-image-1200x726.jpg 992w">
	&lt;div class="container">
		&lt;div class="row mrgn-bttm-md">
			&lt;div class="col-md-7">
			&lt;h1 property="name" id="wb-cont">[Institution name]&lt;/h1>
				&lt;p>Short description of the institution’s mandate.&lt;/p>
				&lt;p>&lt;a class="btn btn-call-to-action" href="#">Super task button [optional]&lt;/a>&lt;/p>
			&lt;/div>
		&lt;/div>
	&lt;/div>
&lt;/div></code></pre>

```

File: components/gc-intro/gc-intro-block-en.html
Comment (duboisp) [reply components/gc-intro/gc-intro-block-en.html:43]: yes, you are right

File: components/gc-intro/gc-intro-block-en.html
Comment (duboisp) [reply components/gc-intro/gc-intro-block-en.html:78]: Good to know, then my comment is not applicable then.

File: components/gc-intro/gc-intro-block-en.html
Comment (duboisp) [inline components/gc-intro/gc-intro-block-en.html:53]: What is the purpose of the CSS grid here?

Is it only to squash the content? If so what do you think if we simply apply the [content width limited](https://wet-boew.github.io/GCWeb/templates/content-page/content-limit-en.html) style on this page? 

Regarding the underline, can we assume it will be replaced by the gc-thickline once it become the new default?

Comment (jmealing) [reply components/gc-intro/gc-intro-block-en.html:53]: The column grid can be changed if needed, depending how people want to display the content. The image is optional.

File: components/gc-intro/gc-intro-block-en.html
Comment (jmealing) [reply components/gc-intro/gc-intro-block-en.html]: Added

File: components/gc-intro/gc-intro-block-en.html
Comment (jmealing) [reply components/gc-intro/gc-intro-block-en.html]: For your other questions:

H1: Let’s come back to it when we know how those other components will work

Info/suggestive: Right now, it has to be followed for the ILP as is in the guidance, but an intro block would be open to customize entirely for other page types until otherwise specified 

Stable component: We think it’s too early to know how we want to address it, its mandatory for the ILP right now, but wouldn’t need to be monitored elsewhere in our opinion

**Comment by duboisp** [comment]
@Garneauma as discussed

**Comment by Garneauma** [comment]
@jmealing Could you please move this documentation in the newly created section /design-patterns ? Thank you.

**Comment by jmealing** [comment]
I can't edit this PR anymore. Not sure why. @delisma ?

Screenshot 2023-12-06 at 1 33 45 PM

**Review by Garneauma — CHANGES_REQUESTED** [review]
- Please wrap the code sample in a container and add a title for it.
- Can you convert the documentation to the new data-first approach? You can base your JSON file on the Date modified component.

File: design-patterns/gc-intro/gc-intro-block-en.html
Comment (Garneauma) [inline design-patterns/gc-intro/gc-intro-block-en.html:2]:
```suggestion
title: Introduction block - documentation
```

**Comment by jmealing** [comment]
@Garneauma I meant to keep this one in draft, not ready yet. Will update Monday

File: design-patterns/gc-intro/gc-intro-block-demos-fr.html
Comment (jmealing) [reply design-patterns/gc-intro/gc-intro-block-demos-fr.html:38]: This is the pattern: https://deploy-preview-307--design-system-canada-ca.netlify.app/common-design-patterns/intro-block

File: design-patterns/gc-intro/gc-intro-block-demos-en.html
Comment (jmealing) [reply design-patterns/gc-intro/gc-intro-block-demos-en.html:38]: The simple example variation. I added a heading to clarify.

File: design-patterns/gc-intro/index.json-ld
Comment (duboisp) [inline design-patterns/gc-intro/index.json-ld:98]: Add a mention that this pattern do require to deactivate the default page container and individual container must be defined through the pages. Something like that.

File: design-patterns/gc-intro/samples/gc-intro-image.html
Comment (duboisp) [inline design-patterns/gc-intro/samples/gc-intro-image.html:5]: Remove the attribute `dir` as by default it is left to right. When user use special text direction, they will apply the lang attribute too on the element.

```suggestion
				<h1 property="name" id="wb-cont">{{ page.title }}</h1>
```

File: design-patterns/gc-intro/samples/gc-intro.html
Comment (duboisp) [inline design-patterns/gc-intro/samples/gc-intro.html:1]:
```suggestion
<h1 property="name" id="wb-cont">{{ page.title }}</h1>
```

It is already defined on the html element and that is an attribute that all sub element inherit

File: design-patterns/gc-intro/index.json-ld
Comment (duboisp) [inline design-patterns/gc-intro/index.json-ld:257]: May be inside the style or a new category, let specify the image ratio and minimal size.

File: design-patterns/gc-intro/index.json-ld
Comment (duboisp) [inline design-patterns/gc-intro/index.json-ld:255]: Humm, this is not a component, it should be only suggestive, we should not record the arrangement.

Although, we can probably add some informal notes regarding the design pattern. That presentation should be different compared to other fully documented component.

**Review by duboisp — CHANGES_REQUESTED** [review]

File: design-patterns/gc-intro/samples/gc-intro-image.html
Comment (duboisp) [inline design-patterns/gc-intro/samples/gc-intro-image.html:1]:
```suggestion
<div class="bg-center bg-cover" data-bgimg-srcset="img/1x1.png 991w, img/ip-cover-image-1200x726.jpg 992w">
```
We are missing the asset img/1x1.png, but should we use that workaround or to update the bgimg-srcset to support a state where there is "no image" for the specified view port?

**Review by duboisp — CHANGES_REQUESTED** [review]
Almost ready, only have a question regarding the 1x1 background image.

File: design-patterns/gc-intro/samples/gc-intro-image.html
Comment (Garneauma) [reply design-patterns/gc-intro/samples/gc-intro-image.html]: Added [PR ](https://github.com/wet-boew/wet-boew/pull/9745) which the current PR is now dependent on.

**Comment by Garneauma** [comment]
The current PR is now dependent on [#9745](https://github.com/wet-boew/wet-boew/pull/9745) being merged.

File: sites/layouts/documentation_pattern.html
Comment (duboisp) [inline sites/layouts/documentation_pattern.html:237]: There is no stability indicator because a design pattern out of scope of our versioning api

```suggestion
```

File: sites/layouts/documentation_pattern.html
Comment (duboisp) [inline sites/layouts/documentation_pattern.html:469]: There is no stability indicator because a design pattern out of scope of our versioning api

```suggestion
```

File: sites/layouts/documentation_pattern.html
Comment (duboisp) [inline sites/layouts/documentation_pattern.html:521]: Please translate in French
```suggestion
								<p>{% if lng == "fr" %}Considération techniques {% else %}Technical considerations{% endif %}:</p>
```

File: design-patterns/gc-intro/index.json-ld
Comment (duboisp) [inline design-patterns/gc-intro/index.json-ld:20]: There is no stability indicator and no version because a design pattern out of scope of our versioning api

```suggestion
```

File: design-patterns/gc-intro/index.json-ld
Comment (duboisp) [inline design-patterns/gc-intro/index.json-ld:67]: There is no stability indicator because a design pattern out of scope of our versioning api

```suggestion
```

File: design-patterns/gc-intro/index.json-ld
Comment (duboisp) [inline design-patterns/gc-intro/index.json-ld:98]: There is no stability indicator because a design pattern out of scope of our versioning api

```suggestion
```

**Review by duboisp — CHANGES_REQUESTED** [review]
Reviewed and tested, everything is functional and do work as expected. I tested with the feature addition that was made to the background image component.

There is a few request of change, see the inline comments:
* Remove any reference that leverage our versioning API
* Make one little translation.

File: design-patterns/gc-intro/index.json-ld
Comment (duboisp) [inline design-patterns/gc-intro/index.json-ld:56]: Don't need to list the CSS utilities, but it worth to list the component/plugin because that support the intended behaviour of this design pattern.

```suggestion
		"data-bgimg-srcset"
```

**Review by duboisp — CHANGES_REQUESTED** [review]
One little and tiny change to make. Everything else is good to go.

Side note: If this PR is merged after gc-institution PR, we will simply need to resolve the conflict on the documentation_pattern.html layout.

**Review by duboisp — DISMISSED** [review]
Reviewed and tested, everything look great.

We will wait for the completion of all the request in https://github.com/orgs/wet-boew/projects/4/views/1 + approval from DTO and PP before to merge it.

**Review by Garneauma — DISMISSED** [review]

**Review by duboisp — APPROVED** [review]
Reviewed again and tested again locally,

No issue and this is ready for merge when approved/coordinated with the other related PRs.

**Review by duboisp — APPROVED** [review]
Nothing has changes, this is only resolving the merge conflict regarding our generated files.

So my last review is still accurate and we did get the approval from PP and DTO to merge the PR related to ILP.

**Pull request merged**
